test/BaseGenReferral.t.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "forge-std/Test.sol";
+import "../src/BaseGen.sol";
+
+contract BaseGenReferralTest is Test {
+    BaseGen public instance;
+    uint256 public constant pricePerMint = 0.0015 ether;
+    uint256 public constant quantity = 10;
+    address public initialOwner;
+    address public receiver;
+    uint256 public maxSupply = 20;
+
+    function setUp() public {
+        initialOwner = vm.addr(1);
+        receiver = vm.addr(2);
+        string memory name = "Generative";
+        string memory symbol = "GEN";
+        string memory contractURI = "ipfs://";
+        string memory baseURI = "https://data.kodadot.xyz/base/"; // suffixed with /?hash=
+        instance = new BaseGen(initialOwner, name, symbol, contractURI, baseURI, maxSupply, receiver);
```

Comment:
Could you guide me in the difference between initialOwner and royalty receiver?
Shouldn't that be one entity?

Reply:
InitialOwner is the provider (Koda) and receiver is the creator of the art. The logic without referrer was that payment is split in half between Koda and artist.